source/frontend/src/components/common/mapFSM/useMapSearch.tsx

```diff
@@ -0,0 +1,204 @@
+import { FeatureCollection, Geometry } from 'geojson';
+import { useCallback } from 'react';
+import { toast } from 'react-toastify';
+
+import { IGeoSearchParams } from '@/constants/API';
+import { useFullyAttributedParcelMapLayer } from '@/hooks/repositories/mapLayer/useFullyAttributedParcelMapLayer';
+import { usePimsPropertyLayer } from '@/hooks/repositories/mapLayer/usePimsPropertyLayer';
+import useKeycloakWrapper from '@/hooks/useKeycloakWrapper';
+import { useModalContext } from '@/hooks/useModalContext';
+import { PMBC_FullyAttributed_Feature_Properties } from '@/models/layers/parcelMapBC';
+import { PIMS_Property_Location_View } from '@/models/layers/pimsPropertyLocationView';
+
+import {
+  emptyFeatureData,
+  emptyFullyFeaturedFeatureCollection,
+  emptyPimsFeatureCollection,
+  MapFeatureData,
+} from './models';
+
+export const useMapSearch = () => {
+  const fullyAttributedService = useFullyAttributedParcelMapLayer();
+  const pimsPropertyLayerService = usePimsPropertyLayer();
+
+  const { setModalContent, setDisplayModal } = useModalContext();
+  const keycloak = useKeycloakWrapper();
+  const logout = keycloak.obj.logout;
+
+  const pimsExecute = pimsPropertyLayerService.loadPropertyLayer.execute;
+  const fullyAttributedServiceFindByPin = fullyAttributedService.findByPin;
+  const fullyAttributedServiceFindByPid = fullyAttributedService.findByPid;
+
+  const fullyAttributedServiceFindOne = fullyAttributedService.findOne;
+  const pimsPropertyLayerServiceFindOne = pimsPropertyLayerService.findOne;
+
+  const searchOneLocation = useCallback(
+    async (latitude: number, longitude: number) => {
+      debugger;
+      let result: MapFeatureData = emptyFeatureData;
+      try {
+        const findOneParcelTask = fullyAttributedServiceFindOne({
+          lat: latitude,
+          lng: longitude,
+        });
+        const findOnePropertyTask = pimsPropertyLayerServiceFindOne(
+          {
+            lat: latitude,
+            lng: longitude,
+          },
+          'GEOMETRY',
+        );
+        const parcelFeature = await findOneParcelTask;
+        const pimsPropertyFeature = await findOnePropertyTask;
+
+        //if found in inventory return or else non inventory
+        const foundGeometry = pimsPropertyFeature
```

Comment:
should this be pimsPropertyFeature?.geometry ?? parcelFeature?.geometry

Reply:
actually, what is the point of this foundGeometry thing? don't the existing if/else on 65 and 67 work just fine?

Reply:
I was matching the functionality as it was before. cleaner now